smarts/core/mission_planner.py

```diff
@@ -296,15 +296,34 @@ def cut_in_waypoints(self, sim, pose: Pose, vehicle, base_waypoint_generator):
             trajectory.append(wp)
         return [trajectory]
 
-    def uturn_waypoints(self, sim, pose: Pose, vehicle):
+    def uturn_waypoints(self, sim, pose: Pose, vehicle, initial_speed):
         # TODO: 1. Need to revisit the approach to calculate the U-Turn trajectory.
         #       2. Wrap this method in a helper.
+
+        ego_position = pose.position[:2]
+        ego_lane = self._road_network.nearest_lane(ego_position)
+        ego_wps = self._waypoints.waypoint_paths_on_lane_at(
+            ego_position, ego_lane.getID(), 60
+        )
+        ego_wps_des_speed = []
+        for px in range(len(ego_wps[0])):
```

Comment:
I had to re-create the waypoint since the frozen is `True` for waypoint class

Reply:
Yes, I see the constraints. Ideally, should be able to specify a desired speed when generating waypoints.